test/sqllogictest/role_create.slt

```diff
@@ -0,0 +1,148 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Loosely based on https://github.com/postgres/postgres/blob/master/src/test/regress/expected/create_role.out
+
+mode cockroach
+
+reset-server
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_ld_rbac_checks TO true;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_rbac_checks TO true;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE ROLE regress_role_admin CREATEDB CREATECLUSTER CREATEROLE;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+CREATE DATABASE regression
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+GRANT CREATE ON DATABASE regression TO regress_role_admin;
+----
+COMPLETE 0
+
+simple conn=mz_system,user=mz_system
+GRANT CREATE ON DATABASE regression TO regress_role_admin WITH GRANT OPTION;
+----
+db error: ERROR: Expected end of statement, found WITH
+
+simple conn=mz_system,user=mz_system
+CREATE ROLE regress_role_limited_admin CREATEROLE;
```

Comment:
There do not seem to be many queries that check what happens after this and all following `CREATE ROLE` succeed. This could include:
- check that the mz_* tables show the role and its atrributes
- attempt to run SQL statements as that role (it seems you have this only for `regress_role_limited_admin` ). In case of success, check the properties of the object created; in case of failure, confirm that it fails with the right error message.